qiskit/visualization/latex.py

```diff
@@ -216,10 +220,14 @@ def _initialize_latex_array(self, aliases=None):
                                     "_{" + str(self.ordered_regs[i].index) + "}" + \
                                     ": 0}"
             else:
-                self._latex[i][0] = "\\lstick{" + \
-                                    self.ordered_regs[i].register.name + "_{" + \
-                                    str(self.ordered_regs[i].index) + "}" + \
-                                    ": \\ket{0}}"
+                if self.layout is None:
+                    self._latex[i][0] = "\\lstick{{ {}_{} : \\ket{{0}} }}".format(
+                        self.ordered_regs[i].register.name, self.ordered_regs[i].index)
+                else:
+                    self._latex[i][0] = "\\lstick{{({}_{})~q_{} : \\ket{{0}} }}".format(
+                        self.ordered_regs[i].register.name,
+                        self.ordered_regs[i].index,
+                        self.layout[self.ordered_regs[i]])
```

Comment:
So this will fail if `self.ordered_regs[i]` is not in the layout for example I wrote a quick test script to verify things locally:

```
from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister

qr = QuantumRegister(2)
cr = ClassicalRegister(2)
qc = QuantumCircuit(qr, cr)
qc.h(qr[0])
qc.cx(qr[0],qr[1])
qc.y(qr[0])
qc.x(qr[1])
qc.measure(qr, cr)
qc.draw(output='latex', filename='pre.png')

from qiskit.compiler import transpile

coupling_map = [[1, 0], [1, 2], [2, 3], [4, 3], [4, 10], [5, 4],
                        [5, 6], [5, 9], [6, 8], [7, 8], [9, 8], [9, 10],
                        [11, 3], [11, 10], [11, 12], [12, 2], [13, 1],
                        [13, 12]]

transpile(qc, basis_gates=['u1', 'u2', 'u3', 'cx', 'id'], coupling_map=coupling_map).draw(output='latex', filename='post.png')
```

and this fails with ``KeyError: "The item Qubit(QuantumRegister(14, 'q'), 0) does not exist in the Layout"``

Reply:
ill have a look

Reply:
The problem is that, after transpilation, the layout does not match the circuit qregs.

Reply:
The big realization here (after talking a bit with @kdk) is that `circuit._layout` only makes sense in "physical" circuits. i.e. circuits with a flat `q` qreg. Updating the PR.